### PR TITLE
[DM] core/components/button.javascript-api

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- update to documentalist@0.0.5. let the [release notes](https://github.com/palantir/documentalist/releases/tag/release-0.0.5) serve as a preview of the changes in this PR.
- 🔥 sweet url-like routes of the form `core/components/button.javascript-api`
- the `nav` tree logic is now built into DM so the gulp task is dead simple config, as it should be

### CODE REVIEW

File: packages/docs/src/index.tsx [hunk 1]
```diff
@@ -40,46 +41,37 @@ const versions = require<string[]>("./generated/versions.json")
...
-                <span className="pt-icon-standard pt-icon-link" />
-            </a>,
-            header,
-        )
-    );
+};
 
-function renderHeading(depth: number) {
-    return (heading: string, key: React.Key, page: IPageData): JSX.Element => {
-        const ref = (depth === 1 ? page.reference : slugify(page.reference, heading));
```
Comment (giladgray) [inline packages/docs/src/index.tsx]: 🎆 no need to re-derive references since we've got `route` now

**Comment by giladgray** [comment]
the current solution simply slugifies the heading text directly with no support for customization, so `.js` is not possible. i'm fine with this approach because it's dead-simple. may be less type-able, but who's really gonna type these out?

**Comment by adidahiya** [comment]
the URLs are a little weird... I've never seen slashes after a hash in the URL like this `docs/dist/index.html#datetime/datetimepicker`... not strictly blocking but are there any alternatives?

**Comment by themadcreator** [comment]
I disagree, i think the slashes in the hashroute are exactly the right format because we are reproducing normal hierarchical url routing within the single page docs app.

**Pull request merged**